ui/page/staking/purchase_modal.go

```diff
@@ -339,6 +339,8 @@ func (tp *stakingModal) calculateTotals() {
 }
 
 func (tp *stakingModal) Handle() {
+	tp.modal.SetDisabled(tp.isLoading)
```

Comment:
Where is the value of `tp.isLoading` modified? That's where `tp.modal.SetDisabled` should be called too.